### PR TITLE
UGENE-6990. Sort MSA rows by leading gap size

### PR DESCRIPTION
**Pull request by yalgaer**

### CODE REVIEW

File: src/corelibs/U2Core/src/datatype/msa/MultipleAlignment.cpp [hunk 1]
```diff
@@ -178,14 +168,46 @@ static bool isLessByLength(const MultipleAlignmentRow &row1, const MultipleAlign
...
-void MultipleAlignmentData::sortRowsByLength(MultipleAlignment::Order order, const U2Region &range) {
+/** Returns leading gap size of the given row. */
+static int getLeadingGapSize(const MultipleAlignmentRow &row) {
+    const U2MsaRowGapModel &gapModel = row->getGapModel();
+    if (gapModel.isEmpty()) {
+        return 0;
+    }
+    const U2MsaGap &gap = gapModel.first();
+    return gap.offset == 0 ? gap.gap : 0;
+}
```
Comment (atiunov) [inline src/corelibs/U2Core/src/datatype/msa/MultipleAlignment.cpp]: Move it to MultipleAlignmentRowData class.

File: src/corelibs/U2View/src/ov_msa/MSAEditor.cpp [hunk 2]
```diff
@@ -755,18 +767,14 @@ void MSAEditor::sl_showCustomSettings() {
     AppContext::getAppSettingsGUI()->showSettingsDialog(ColorSchemaSettingsPageId);
 }
 
-void MSAEditor::sortSequences(bool isByName, const MultipleAlignment::Order &sortOrder) {
+void MSAEditor::sortSequences(const MultipleAlignment::SortType &sortType, const MultipleAlignment::Order &sortOrder) {
```
Comment (atiunov) [inline src/corelibs/U2View/src/ov_msa/MSAEditor.cpp:770]: Enums a too simple types to pass them as const references.

**Review by atiunov — CHANGES_REQUESTED** [review]

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.cpp [hunk 3]
```diff
@@ -6713,7 +6713,48 @@ GUI_TEST_CLASS_DEFINITION(test_6981) {
     QString annTreeItem = GTUtilsAnnotationsTreeView::getSelectedItem(os);
     CHECK_SET_ERR(annTreeItem == "ann", QString("Incorrect selected item name, expected: ann, current: %1").arg(annTreeItem));
 }
+GUI_TEST_CLASS_DEFINITION(test_6990) {
+    // Open COI.aln
+    GTFileDialog::openFile(os, dataDir + "samples/CLUSTALW", "COI.aln");
+    GTUtilsMsaEditor::checkMsaEditorWindowIsActive(os);
+
+    // Sort -> Leading gap
```
Comment (atiunov) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.cpp]: Test comments should be more clear.

**Review by atiunov — CHANGES_REQUESTED** [review]
Add a test that uses context menu to sort.
Change code at MsaEditorSortSequencesWidget:76: use combobox item data instead of items order. Order could be changed for some reasons.

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.cpp (hunk 3 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.cpp]: Fixed

File: src/corelibs/U2View/src/ov_msa/MSAEditor.cpp (hunk 2 above)
Comment (yalgaer) [reply src/corelibs/U2View/src/ov_msa/MSAEditor.cpp:770]: Fixed

File: src/corelibs/U2Core/src/datatype/msa/MultipleAlignment.cpp (hunk 1 above)
Comment (yalgaer) [reply src/corelibs/U2Core/src/datatype/msa/MultipleAlignment.cpp]: Fixed

**Comment by yalgaer** [comment]
> Change code at MsaEditorSortSequencesWidget:76: use combobox item data instead of items order. Order could be changed for some reasons.

This is out of the scope of the current change (these lines were not touched): I will create a separate bug for that.

**Review by atiunov — DISMISSED** [review]

**Comment by yalgaer** [comment]
> Add a test that uses context menu to sort.

The test was added

**Review by atiunov — APPROVED** [review]

**Pull request merged**